x-pack/plugins/security/public/views/management/management.js

```diff
@@ -66,16 +66,17 @@ routes.defaults(/\/management/, {
         }
       }
 
-      deregisterSecurity();
-      if (!showSecurityLinks) return;
-
-      // getCurrent will reject if there is no authenticated user, so we prevent them from seeing the security
-      // management screens
-      //
-      // $promise is used here because the result is an ngResource, not a promise itself
-      return ShieldUser.getCurrent().$promise
-        .then(ensureSecurityRegistered)
-        .catch(deregisterSecurity);
+      if (!showSecurityLinks) {
```

Comment:
I don't recall why `deregisterSecurity()` was previously always being called, I assume it was for some weird edge-case. Are we confident that we want to change this now?

Reply:
@kobelb This code is called every time a management route is requested. Every time the route is requested the security section is being deregistered and reregistered. This creates an _extremely_ noisy registration system which is difficult to understand let alone present to the user without weird side effects. (mainly the disappearance and reappearance of items) 

Previously this didn't matter because management sections were rendered synchronously. Deregister and register before rendering and everything is fine. Beats needed async loading which caused the noise to become visible.

Maybe this was doing something I didn't understand but its likely that it was just a thing that happened and worked until now.

Reply:
This commit provides some explanation of why this was changed initially, but to be honest I'm not following it entirely: https://github.com/elastic/x-pack/commit/d40253b1eee7a957ba956393baf5a4d252ca870c

@Bargs perhaps you could enlighten us?

Reply:
Sorry, I don't know why it was getting called on every route change, I only made sure each registration had a subsequent deregistration. It used to thrown an error if `register` was called twice without a deregister in-between. But I see there's an `ensureSecurityRegistered` function being used now so maybe that's no longer an issue.

Reply:
I withdraw my concerns then :)